core/src/main/java/io/kestra/core/runners/VariableRenderer.java

```diff
@@ -34,9 +35,10 @@
 @Singleton
 public class VariableRenderer {
     private static final Pattern RAW_PATTERN = Pattern.compile("\\{%[-]*\\s*raw\\s*[-]*%\\}(.*?)\\{%[-]*\\s*endraw\\s*[-]*%\\}");
+    public static final int MAX_RENDERING_AMOUNT = 100;
 
-    private Handlebars handlebars;
-    private final PebbleEngine pebbleEngine;
+    public static Handlebars handlebars;
+    public static PebbleEngine pebbleEngine;
```

Comment:
I had no choice but to put them as static since I need to access them in the new static `renderOnce` & `renderRecursively` methods which are also called from RenderFunction (and in such Pebble functions you don't have access to Micronaut context or the whole PebbleEngine)

Reply:
No, your function can themself be CDI beans, this is what we do already for some of them.
Look inside `io.kestra.core.runners.pebble.Extension`

Reply:
That's true my bad